command/operator_snapshot_inspect.go

```diff
@@ -58,20 +114,137 @@ func (c *OperatorSnapshotInspectCommand) Run(args []string) int {
 	}
 	defer f.Close()
 
-	meta, err := snapshot.Verify(f)
+	meta, info, err := inspect(f)
 	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Error verifying snapshot: %s", err))
+		c.Ui.Error(fmt.Sprintf("Error inspecting snapshot: %s", err))
 		return 1
 	}
+	stats := generateStats(info)
+
+	// format as JSON if requested
+	if json {
+		data := SnapshotInspectFormat{
+			Meta:  meta,
+			Stats: stats,
+		}
+		out, err := Format(json, "", data)
+		if err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
+
+		c.Ui.Output(out)
+		return 0
+	}
 
-	output := []string{
+	// print human readable output
+	c.Ui.Output(formatListWithSpaces([]string{
+		fmt.Sprintf("Created|%s", extractTimeFromName(meta.ID)),
 		fmt.Sprintf("ID|%s", meta.ID),
-		fmt.Sprintf("Size|%d", meta.Size),
+		fmt.Sprintf("Size|%s", ByteSize(uint64(meta.Size))),
 		fmt.Sprintf("Index|%d", meta.Index),
 		fmt.Sprintf("Term|%d", meta.Term),
 		fmt.Sprintf("Version|%d", meta.Version),
+	}))
+	c.Ui.Output("")
+
+	output := []string{
+		"Type|Count|Size",
+		"----|-----|----",
+	}
+
+	for _, stat := range stats {
+		output = append(output, fmt.Sprintf("%s|%d|%s", stat.Name, stat.Count, ByteSize(uint64(stat.Sum))))
 	}
+	output = append(output, "----|-----|----")
+	output = append(output, fmt.Sprintf("Total|-|%s", ByteSize(uint64(info.TotalSize))))
 
 	c.Ui.Output(formatList(output))
 	return 0
 }
+
+func inspect(file io.Reader) (*raft.SnapshotMeta, *SnapshotInfo, error) {
+	info := &SnapshotInfo{
+		Stats:     make(map[nomad.SnapshotType]typeStats),
+		TotalSize: 0,
+	}
+
+	// w is closed by CopySnapshot
+	r, w := io.Pipe()
+	cr := &countingReader{wrappedReader: r}
+	errCh := make(chan error)
+	metaCh := make(chan *raft.SnapshotMeta)
+
+	go func() {
+		meta, err := snapshot.CopySnapshot(file, w)
+		if err != nil {
+			errCh <- fmt.Errorf("failed to read snapshot: %w", err)
+		} else {
+			metaCh <- meta
+		}
+	}()
+
+	handler := func(header *nomad.SnapshotHeader, snapType nomad.SnapshotType, dec *codec.Decoder) error {
+		name := snapType.String()
+		stat := info.Stats[snapType]
+
+		if stat.Name == "" {
+			stat.Name = name
+		}
+
+		var val interface{}
+		err := dec.Decode(&val)
+		if err != nil {
+			return fmt.Errorf("failed to decode snapshot type %v, error %v", snapType, err)
```

Comment:
We don't need to specify this is an error which is handled by the UI. I think it's also worth removing the type wording, as this is leaky.
```suggestion
			return fmt.Errorf("failed to decode snapshot %q: %v", snapType, err)
```